src/history.js

```diff
@@ -37,17 +40,77 @@ export class History {
    */
   init_() {
     window.addEventListener('popstate', event => {
-      const urlPath = event.state ? event.state.urlPath : null;
-      this.handleChangeHistoryState_(urlPath);
+      const state = event.state;
+      if (!state) {
+        this.decreaseCurrentStateId();
+        this.handleChangeHistoryState_(true /* isBack */, true /* isLastBack */, null);
+        return;
+      }
+
+      // The popped browser history state id.
+      const poppedStateId = (typeof state.stateId !== 'undefined') ? state.stateId : null;
+      // History index that the AMP doc uses.
+      const poppedStackIndex = (typeof state.stackIndex !== 'undefined') ? state.stackIndex : null;
+
+      const isBack = this.isBack_(poppedStateId);
+      if (isBack) {
+        this.decreaseCurrentStateId();
+      } else {
+        this.increaseCurrentStateId();
+      }
```

Comment:
I'm keeping a counter so that I can compare it with the poppedStateId to know if we're going forward or backwards. If we're going backward, then the viewer can send a 'historyPopped' message to the ampDoc. This is useful when the lightbox or sidebar are opened. The back button should close them.